src/vm_stack.zig

```diff
@@ -5303,24 +5336,34 @@ pub const StackVM = struct {
         }
     }
 
-    pub fn resumeInvoke(vm: *VM, module: *ModuleInstance, returns: []Val) anyerror!void {
+    pub fn resumeInvoke(vm: *VM, module: *ModuleInstance, returns: []Val, opts: ResumeInvokeOpts) anyerror!void {
         var self: *StackVM = fromVM(vm);
 
-        std.debug.assert(self.debug_state != null);
-        std.debug.assert(self.debug_state.?.is_invoking);
-
-        const debug_state = &self.debug_state.?;
-        const opcode: Opcode = blk: {
+        var pc: u32 = 0;
+        const opcode: Opcode = if (self.debug_state) |debug_state| blk: {
```

Comment:
Instead of using these blk blocks, let's have simpler if-else chain and set `opcode` the same way `pc` is set inside the scopes.